### PR TITLE
[integration] Enable integration tests via Github actions

### PR DESCRIPTION
**Pull request by abikouo**
This pull request contains multiple changes

- Add GH workflow for integration tests
- Add integration tests for role `ec2_instance_terminate_by_tag`
- Rename some roles variable to fix `ansible-lint` issues
- Stabilize integration tests 

Integration tests have been tested via #38

### CODE REVIEW

File: .github/workflows/changelog.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
-name: changelog
+name: CI
```
[the remark this answers is not in the document]
Comment (abikouo) [reply .github/workflows/changelog.yml]: updated

File: roles/aws_setup_credentials/tasks/main.yml [hunk 2]
```diff
@@ -1,7 +1,8 @@
 ---
 - name: Define intial value for credentials
   ansible.builtin.set_fact:
-    aws_role_credentials: {}
+    aws_setup_credentials__output: {}
```
[the remark this answers is not in the document]
Comment (abikouo) [reply roles/aws_setup_credentials/tasks/main.yml:4]: Indeed this is to fix ansible-lint issues, there are other related changes. The changelog has been added
I don't know why the test passed 2 months ago, but if you tried to run them now on the `main` branch it would fail

File: roles/customized_ami/defaults/main.yml [hunk 3]
```diff
@@ -7,9 +7,9 @@ source_ami_filters:
   name: Fedora-Cloud-Base-35-*
 source_ami_user_name: ec2-user
 
-custom_ami_vpc_cidr: 10.1.0.0/16
-custom_ami_subnet_cidr: 10.1.0.0/24
-custom_ami_ec2_instance_name: "{{ custom_ami_name }}-ec2"
+custom_ami_vpc_cidr: '173.12.0.0/16'
+custom_ami_subnet_cidr: '173.12.1.0/24'
```
Comment (gravesm) [inline roles/customized_ami/defaults/main.yml:11]: Why this change?

File: roles/customized_ami/templates/fedora-user-data.j2 [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+{% for package in custom_ami_packages %}
+sudo dnf install -y {{ package }}
```
[the remark this answers is not in the document]
Comment (abikouo) [reply roles/customized_ami/templates/fedora-user-data.j2]: updated

File: roles/customized_ami/defaults/main.yml (hunk 3 above)
Comment (abikouo) [reply roles/customized_ami/defaults/main.yml]: updated

File: tests/integration/targets/test_customized_ami/playbooks/validate_ami.yml
[the remark this answers is not in the document]
Comment (gravesm) [reply tests/integration/targets/test_customized_ami/playbooks/validate_ami.yml:1]: What's the problem with this target?

[the remark this answers is not in the document]
Comment (abikouo) [reply tests/integration/targets/test_customized_ami/playbooks/validate_ami.yml:1]: it is hard to test using `t2.micro` instances and for now, AWS CI account does not support creating `t2.large` instances

[the remark this answers is not in the document]
Comment (abikouo) [reply tests/integration/targets/test_customized_ami/playbooks/validate_ami.yml:1]: issue found, integration test targets is fully working now

**Review by gravesm — APPROVED** [review]

**Review by gravesm — CHANGES_REQUESTED** [review]
Woops, didn't mean to approve. Would still like to see the ami tests enabled.

File: tests/integration/targets/test_customized_ami/aliases [hunk 5]
```diff
@@ -1 +1,4 @@
-disabled
+cloud/aws
+role/customized_ami
+time=40m
+disabled
```
[the remark this answers is not in the document]
Comment (abikouo) [reply tests/integration/targets/test_customized_ami/aliases:4]: I forgot to set them as enabled, I will fix it.

**Review by gravesm — APPROVED** [review]
